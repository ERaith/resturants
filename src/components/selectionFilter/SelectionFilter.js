export function SelectionFilter({
  dropDownItems,
  filterTermActive,
  handleChange,
  id,
}) {
  return (
    <div className="filter-container">
      <label htmlFor={id}> {id.toUpperCase()} </label>
      <select key={id} onChange={handleChange} id={id} className="filter">
        <option key={`Default-Option:${id}`} defaultValue>
          All
        </option>
        {dropDownItems.map((item) => {
          return <option key={item}>{item}</option>;
        })}
      </select>
      <input
        checked={filterTermActive}
        className="checkbox"
        id={`${id}:isActive`}
        onChange={handleChange}
        type="checkbox"
      ></input>
    </div>
  );
}
