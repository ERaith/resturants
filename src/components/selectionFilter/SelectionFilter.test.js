import React from "react";
import { SelectionFilter } from "./SelectionFilter";
import { render, fireEvent } from "@testing-library/react";

describe("SelectionFilter", () => {
  describe("Snapshots", () => {
    it("should match the snapshot", () => {
      const { asFragment } = render(
        <SelectionFilter
          id="Test"
          dropDownItems={["one", "two"]}
          handleChange={jest.fn(() => "change")}
          filterTermActive={false}
        />
      );

      expect(asFragment(<SelectionFilter />)).toMatchSnapshot();
    });
  });

  describe("user interactions", () => {
    it("should invoke handleChange when selection is changed", () => {
      const mockHandleChange = jest.fn();
      const { getByRole } = render(
        <SelectionFilter
          id="Test"
          dropDownItems={["one", "two"]}
          handleChange={mockHandleChange}
          filterTermActive={false}
        />
      );

      expect(getByRole("combobox", "#Test")).toBeTruthy();
      expect(getByRole("combobox", "#Test").value).toBe("All");

      fireEvent.change(getByRole("combobox", "#Test"), {
        target: { value: "two" },
      });
      expect(mockHandleChange).toBeCalled();
      expect(getByRole("combobox", "#Test").value).toBe("two");
    });
  });
});
