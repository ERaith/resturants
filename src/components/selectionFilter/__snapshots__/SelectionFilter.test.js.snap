// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectionFilter Snapshots should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="filter-container"
  >
    <label
      for="Test"
    >
       TEST 
    </label>
    <select
      class="filter"
      id="Test"
    >
      <option>
        All
      </option>
      <option>
        one
      </option>
      <option>
        two
      </option>
    </select>
    <input
      class="checkbox"
      id="Test:isActive"
      type="checkbox"
    />
  </div>
</DocumentFragment>
`;
